diff/parse.go

```diff
@@ -103,7 +103,15 @@ func (p *fileParser) parseHunks() ([]*Hunk, error) {
 		return nil, ErrNoHunks
 	}
 	if !bytes.HasPrefix(b, []byte(tokenStartHunk)) {
-		return nil, ErrNoHunks
+		b, err := p.r.Peek(len(tokenDiffGit))
+		if err != nil {
+			return nil, ErrNoHunks
+		}
+		if !bytes.HasPrefix(b, []byte(tokenDiffGit)) {
```

Comment:
Can you invert this condition?

This is special case for "git diff", not general unified diff. something like this.

```go
if bytes.HasPrefix(b, []byte(tokenDiffGit)) {
  // git diff may contain a file diff with empty hunks.
  // e.g. delete an empty file.
  return []*Hunk{}, nil
}
```